osutil/disks/disks_linux.go

```diff
@@ -635,8 +635,23 @@ func (d *disk) populatePartitions() error {
 				continue
 			}
 
-			// then the device is a partition, get the udev props for it
 			partDev := filepath.Base(path)
+
+			// then the device is a partition, trigger any udev events for it
+			// that might be sitting around and then get the udev props for it
+
+			// trigger
+			out, err := exec.Command("udevadm", "trigger", "--name-match="+partDev).CombinedOutput()
+			if err != nil {
+				return osutil.OutputErr(out, err)
+			}
+
+			// then settle
+			out, err = exec.Command("udevadm", "settle", "--timeout=180").CombinedOutput()
```

Comment:
@anonymouse64 it seems that trigger has a --settle option itself? that one doesn't have a timeout OTOH it seems to be targeted instead of just waiting for a steady state, should we consider using that?

Reply:
Unfortunately that option is not available on all systems where we will run this code, namely 16.04/UC16 does not have this option `--settle` with the `trigger command`. Perhaps the implementation unconditionally settles after triggering on 16.04, I'm not sure though

Reply:
given that we don't want to check what version we are on from here? I suppose we should go with the code as is

Reply:
yes I would prefer to not have to check what version of systemd/udev we have at this point, but we could refactor to do that instead if you prefer